platform/default/default_file_source.cpp

```diff
@@ -160,10 +160,16 @@ class DefaultFileSource::Impl {
                     revalidation.priorExpires = offlineResponse->expires;
                     revalidation.priorEtag = offlineResponse->etag;
                     callback(*offlineResponse);
-                }
 
+                }
             }
 
+            auto now = util::now();
+            
+            if (bool(revalidation.priorExpires) && *revalidation.priorExpires > now) {
+              return;
+            }
```

Comment:
only immediately request a web tile after fetching a tile from the database if it is expired